### PR TITLE
Fixes #33952 - do not audit internal columns

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Disable auditing of Katello internal column.

Found as https://github.com/Katello/katello/pull/9270#issuecomment-972763008

#### What are the changes introduced in this pull request?
Stop auditing internal column `created_in_katello`

#### What are the testing steps for this pull request?
Create org

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
@jturel I think this would be the fix, tho not tested and just an opinion :)

**Comment by jturel** [comment]
[test katello]

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by ezr-ondrej** [comment]
[test katello]
I mean... what is going on with that pipeline?

**Comment by jturel** [comment]
[test katello]

**Review by jeremylenz — COMMENTED** [review]
I'm not sure if I'm doing something wrong in my testing, but I created orgs both in master and with this PR checked out.  They both seem to produce 5 new audit records, none of which is an 'update'. And it seems to be the same.  what am I missing?..

new_org_audits

**Comment by jturel** [comment]
[test katello]

**Comment by jturel** [comment]
@ezr-ondrej this pr seems like a special case for some reason. can you rebase it from master and see if that somehow helps?

**Comment by ezr-ondrej** [comment]
> I'm not sure if I'm doing something wrong in my testing, but I created orgs both in master and with this PR checked out.

I believe this is happening for the Default seeded org? not sure, I can ask the QE

**Comment by ezr-ondrej** [comment]
> can you rebase it from master and see if that somehow helps?

I'm trying now, but this error is happening quite a lot in core as well, usually restarting the tests helps, but it definitelly does happen. I do not know where to start debugging it, but if I could do something about it, I'd gladly do, it's bit annoying.

**Comment by jeremylenz** [comment]
> usually restarting the tests helps

Is this a QE test that's failing? Can I see the failure output anywhere?

**Comment by jeremylenz** [comment]
oh, you mean the failures on this PR?
```
PG::ConnectionBad: FATAL:  database "test-katello-pr-test-9446-dev" does not exist
```
yeah.. no idea :confused:

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by jturel** [comment]
Looking more closely at the jenkins stack trace it gives the impression the error is related to this very PR:

```
[2021-11-23T22:19:20.970Z] rake aborted!

[2021-11-23T22:19:20.970Z] ActiveRecord::NoDatabaseError: FATAL:  database "test-katello-pr-test-9450-1-dev" does not exist

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/postgresql_adapter.rb:50:in `rescue in postgresql_connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/postgresql_adapter.rb:33:in `postgresql_connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:887:in `new_connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:931:in `checkout_new_connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:910:in `try_to_checkout_new_connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:871:in `acquire_connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:593:in `checkout'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:437:in `connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:1119:in `retrieve_connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_handling.rb:221:in `retrieve_connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/connection_handling.rb:189:in `connection'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/model_schema.rb:340:in `table_exists?'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/attribute_methods/primary_key.rb:91:in `get_primary_key'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/attribute_methods/primary_key.rb:79:in `reset_primary_key'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/attribute_methods/primary_key.rb:67:in `primary_key'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/attribute_methods/primary_key.rb:81:in `reset_primary_key'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activerecord-6.0.3.7/lib/active_record/attribute_methods/primary_key.rb:67:in `primary_key'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/audited-4.10.0/lib/audited/auditor.rb:439:in `default_ignored_attributes'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/audited-4.10.0/lib/audited/auditor.rb:457:in `calculate_non_audited_columns'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/audited-4.10.0/lib/audited/auditor.rb:378:in `non_audited_columns'

[2021-11-23T22:19:20.970Z] /home/jenkins/workspace/katello-pr-test@2/app/models/katello/concerns/organization_extensions.rb:8:in `block in <module:OrganizationExtensions>'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activesupport-6.0.3.7/lib/active_support/concern.rb:122:in `class_eval'

[2021-11-23T22:19:20.970Z] /usr/local/rvm/gems/ruby-2.7.0@katello-pr-test-9450-1/gems/activesupport-6.0.3.7/lib/active_support/concern.rb:122:in `append_features'

[2021-11-23T22:19:20.970Z] /home/jenkins/workspace/katello-pr-test@2/lib/katello/engine.rb:176:in `include'

[2021-11-23T22:19:20.970Z] /home/jenkins/workspace/katello-pr-test@2/lib/katello/engine.rb:176:in `block in <class:Engine>'

```

is non_audited_columns trying to read from the DB before it's even created?

**Comment by ezr-ondrej** [comment]
> is non_audited_columns trying to read from the DB before it's even created?

indeed :facepalm: tho having this error repeated so many times, I just assumed it's the one xD the getter can not be called because it goes to the DB, but I do not want to override the default behaviour, so I went with a method override.

**Comment by ezr-ondrej** [comment]
:green_apple:

**Comment by jeremylenz** [comment]
@ezr-ondrej any tips on how to reproduce the condition? (see my [comment above](https://github.com/Katello/katello/pull/9798#pullrequestreview-812705071))

**Comment by ezr-ondrej** [comment]
> @ezr-ondrej any tips on how to reproduce the condition? (see my [comment above](https://github.com/Katello/katello/pull/9798#pullrequestreview-812705071))

Hmm in your snapshot I do not see any Audit related to the Org itself, only creation of followup resources. The Audit of the org itself will probably be visible only in Any context (possibly?)?

**Comment by jeremylenz** [comment]
> The Audit of the org itself will probably be visible only in Any context

hmm, I tried the Any context (it seems to select that for you) and still see the same audits..

**Comment by jeremylenz** [comment]
@ezr-ondrej Which dev setup are you using? perhaps it's a difference between them :thinking:

**Comment by ezr-ondrej** [comment]
Well there can. I'm not sure how that actually happens neighter, but these changes make sense IMHO.
If nothing else, we will not save the record twice and we will not audit internal columns :)

Plus I'm sure the QE filing the attached BZ, doesn't lie :thinking: :)

File: app/services/katello/organization_creator.rb [hunk 1]
```diff
@@ -45,9 +46,6 @@ def create!
         seed!
 
         create_backend_objects!
-
-        @organization.created_in_katello = true
```
Comment (jeremylenz) [inline app/services/katello/organization_creator.rb:49]: I think `created_in_katello` is meant to track whether the Candlepin owner and library environment are correctly in place. Setting `created_in_katello` to true before running `create_backend_objects` will defeat the purpose, no?  Let's ask the original author @jturel for more comment.  Sorry I am being so particular but I wouldn't want to re-break this especially without testing it :wink:

Comment (ezr-ondrej) [reply app/services/katello/organization_creator.rb:49]: Sure, no worries :+1:  If you have concerns, let's solve those first :+1:

Comment (jturel) [reply app/services/katello/organization_creator.rb:49]: It all happens in a transaction so there should be no functional difference - no test failure leads me to believe that especially. I probably put the assignment to created_in_katello to true where I did per Jeremy's comment above - only setting true after the Candlepin and other backend parts have happened - so it's more of a conceptual thing.

**Comment by jturel** [comment]
I want to avoid a downstream rebase so please hold merging this until https://github.com/Katello/katello/pull/9822 has been merged. This will need a rebase at that point but imo that's better than doing it downstream :)

**Comment by jeremylenz** [comment]
Based on the comments above I'm OK with merging once #9822 is in.

**Comment by jturel** [comment]
ready for rebase here!

**Comment by ezr-ondrej** [comment]
Rebased :)

**Review by jeremylenz — APPROVED** [review]
Thanks @ezr-ondrej !

ACK based on above comments :+1:

**Comment by evgeni** [comment]
Could this have broken Proxy syncs?

The last successful nightly was with efb05fd, the first failing is 131825e.

The error looks like this:
```
  2022-01-13T00:39:13 [I|app|79b72ce3] Started POST "/katello/api/capsules/2/content/sync" for 192.168.121.132 at 2022-01-13 00:39:13 +0000
  2022-01-13T00:39:13 [I|app|79b72ce3] Processing by Katello::Api::V2::CapsuleContentController#sync as JSON
  2022-01-13T00:39:13 [I|app|79b72ce3]   Parameters: {"api_version"=>"v2", "id"=>"2", "capsule_content"=>{}}
  2022-01-13T00:39:13 [I|app|79b72ce3] Authorized user admin(Admin User)
  2022-01-13T00:39:13 [I|bac|79b72ce3] Task {label: , execution_plan_id: 61b4d94e-403c-4523-afa1-b437ecff32d9} state changed: pending
  2022-01-13T00:39:13 [I|bac|79b72ce3] Task {label: Actions::Katello::CapsuleContent::Sync, id: eabdcad7-9cf8-410f-a7bb-58c992d2825c, execution_plan_id: 61b4d94e-403c-4523-afa1-b437ecff32d9} state changed: planning
  2022-01-13T00:39:14 [E|bac|79b72ce3] Katello::Resources::Candlepin::Owner: 404 Not Found {"displayMessage":"Organization with id Default_Organization could not be found.","requestUuid":"ddfc3f6a-a1f8-4f9e-9a7f-354cbdf09d3e"} (POS  T /candlepin/owners/Default_Organization/uebercert) (RestClient::NotFound)
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.2/lib/restclient/abstract_response.rb:223:in `exception_with_response'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.2/lib/restclient/abstract_response.rb:103:in `return!'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.2/lib/restclient/request.rb:809:in `process_result'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.2/lib/restclient/request.rb:725:in `block in transmit'
   79b72ce3 | /opt/rh/rh-ruby27/root/usr/share/ruby/net/http.rb:933:in `start'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.2/lib/restclient/request.rb:715:in `transmit'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.2/lib/restclient/request.rb:145:in `execute'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.2/lib/restclient/request.rb:52:in `execute'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.2/lib/restclient/resource.rb:67:in `post'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/lib/katello/http_resource.rb:92:in `issue_request'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/lib/katello/http_resource.rb:47:in `block (2 levels) in singleton class'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/lib/katello/resources/candlepin/owner.rb:94:in `generate_ueber_cert'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/models/katello/glue/candlepin/owner.rb:56:in `generate_debug_cert'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/models/katello/glue/candlepin/owner.rb:62:in `rescue in load_debug_cert'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/models/katello/glue/candlepin/owner.rb:59:in `load_debug_cert'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/models/katello/glue/candlepin/owner.rb:14:in `block (2 levels) in included'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/lib/katello/lazy_accessor.rb:118:in `instance_eval'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/lib/katello/lazy_accessor.rb:118:in `run_initializer'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/lib/katello/lazy_accessor.rb:97:in `lazy_attribute_get'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/lib/katello/lazy_accessor.rb:44:in `block (2 levels) in lazy_accessor'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/services/cert/certs.rb:4:in `ueber_cert'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/services/cert/certs.rb:40:in `verify_ueber_cert'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/models/katello/concerns/smart_proxy_extensions.rb:418:in `block in verify_ueber_certs'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/activerecord-6.0.3.7/lib/active_record/relation/delegation.rb:87:in `each'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/models/katello/concerns/smart_proxy_extensions.rb:417:in `verify_ueber_certs'
   79b72ce3 | /opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.4.0.pre.master/app/lib/actions/katello/capsule_content/sync.rb:23:in `plan'
```

And looking at the `candlepin.log`s, the failing one has additional `POST`s to `/candlepin/owners/Default_Organization/uebercert`, which are absent from the green one.

**Pull request merged**